mysql-ballerina/Module.md

```diff
@@ -0,0 +1,106 @@
+in the `jdbc:Client`, you can pass it without named params.
+
+The `dbClient3` uses the named params to pass the attributes since it is skipping some params in the constructor.
+Further [mysql:Options](https://ballerina.io/learn/api-docs/ballerina/#/mysql/records/Options)
+is passed to configure the SSL and connection timeout in the MySQL client.
+
+Similarly, the `dbClient4` uses the named params and it provides an unshared connection pool in the type of
+[sql:ConnectionPool](https://ballerina.io/learn/api-docs/ballerina/#/sql/records/ConnectionPool)
+to be used within the client.
+For more details about connection pooling, see the [SQL Module](https://ballerina.io/learn/api-docs/ballerina/#/sql).
+
+```ballerina
+mysql:Client|sql:Error dbClient1 = new ();
+mysql:Client|sql:Error dbClient2 = new ("localhost", "rootUser", "rooPass", 
+                              "information_schema", 3306);
+                              
+mysql:Options mysqlOptions = {
+  ssl: {
+    mode: mysql:SSL_PREFERRED
+  },
+  connectTimeout: 10
+};
+mysql:Client|sql:Error dbClient3 = new (user = "rootUser", password = "rootPass",
+                              options = mysqlOptions);
+mysql:Client|sql:Error dbClient4 = new (user = "rootUser", password = "rootPass",
+                              connectionPool = {maxOpenConnections: 5});
+```
+
+#### Using SSL
+In order to connect the MySQL database using SSL connection, you must add SSL configurations to `mysql:Options` when crating the `dbClient`.
+For SSLMode, You can select one of the modes `mysql:SSL_PREFERRED`, `mysql:SSL_REQUIRED`, `mysql:SSL_VERIFY_CA`, or `mysql:SSL_VERIFY_IDENTITY` according to the requirement.
+for key and cert files, you must provide the files in .p12 format.
+
+```ballerina
+string clientStorePath = "/path/to/keystore.p12";
+string turstStorePath = "/path/to/truststore.p12";
+
+mysql:Options mysqlOptions = {
+  ssl: {
+    mode: mysql:SSL_PREFERRED,
+    key: {
+        path: clientStorePath,
+        password: "password"
+    },
+    cert: {
+        path: turstStorePath,
+        password: "password"
+    }
+  }
+};
+```
+
+You can find more details about each property in the
+[mysql:Client](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) constructor.
+
+The [mysql:Client](https://ballerina.io/learn/api-docs/ballerina/#/mysql/clients/Client) references
+[sql:Client](https://ballerina.io/learn/api-docs/ballerina/#/sql/abstractObjects/Client) and all the operations
+defined by the `sql:Client` will be supported by the `mysql:Client` as well.
+
+For more information on all the operations supported by the `mysql:Client`, which include the below, see the [SQL Module](https://ballerina.io/learn/api-docs/ballerina/#/sql).
+
+1. Connection Pooling
+2. Querying data
+3. Inserting data
+4. Updating data
+5. Deleting data
+6. Batch insert and update data
+7. Execute stored procedures
+8. Closing client
```

Comment:
Shall we add small explanations with code snippets

Reply:
this is resolved in https://github.com/ballerina-platform/module-ballerinax-mysql/pull/157/commits/ca750df90affd7d7c591c04ff542c9f307a82f8b